contracts/asset-proxy/contracts/src/bridges/UniswapBridge.sol

```diff
@@ -0,0 +1,221 @@
+            fromTokenAddress,
+            toTokenAddress
+        );
+        // Grant an allowance to the exchange.
+        _grantExchangeAllowance(state.exchange);
+        // Get our balance of `fromTokenAddress` token.
+        state.fromTokenBalance = IERC20Token(fromTokenAddress).balanceOf(address(this));
+        // Get the weth contract.
+        state.weth = _getWethContract();
+
+        // Convert from WETH to a token.
+        if (fromTokenAddress == address(state.weth)) {
+            // Unwrap the WETH.
+            state.weth.withdraw(state.fromTokenBalance);
+            // Buy as much of `toTokenAddress` token with ETH as possible and
+            // transfer it to `to`.
+            state.exchange.ethToTokenTransferInput.value(state.fromTokenBalance)(
+                // No minimum buy amount.
+                0,
+                // Expires after this block.
+                block.timestamp,
+                // Recipient is `to`.
+                to
+            );
+
+        // Convert from a token to WETH.
+        } else if (toTokenAddress == address(state.weth)) {
+            // Buy as much ETH with `fromTokenAddress` token as possible.
+            uint256 ethBought = state.exchange.tokenToEthSwapInput(
+                // Sell all tokens we hold.
+                state.fromTokenBalance,
+                // No minimum buy amount.
+                0,
+                // Expires after this block.
+                block.timestamp
+            );
+            // Wrap the ETH.
+            state.weth.deposit.value(ethBought)();
+            // Transfer the WETH to `to`.
+            IERC20Token(toTokenAddress).transfer(to, ethBought);
+
+        // Convert from one token to another.
+        } else {
+            // Buy as much `toTokenAddress` token with `fromTokenAddress` token
+            // and transfer it to `to`.
+            state.exchange.tokenToTokenTransferInput(
+                // Sell all tokens we hold.
+                state.fromTokenBalance,
+                // No minimum buy amount.
+                0,
+                // No minimum intermediate ETH buy amount.
+                0,
+                // Expires after this block.
+                block.timestamp,
+                // Recipient is `to`.
+                to,
+                // Convert to `toTokenAddress`.
+                toTokenAddress
+            );
+        }
+        return BRIDGE_SUCCESS;
+    }
+
+    /// @dev `SignatureType.Wallet` callback, so that this bridge can be the maker
+    ///      and sign for itself in orders. Always succeeds.
+    /// @return magicValue Success bytes, always.
+    function isValidSignature(
+        bytes32,
+        bytes calldata
+    )
+        external
+        view
+        returns (bytes4 magicValue)
+    {
+        return LEGACY_WALLET_MAGIC_VALUE;
+    }
+
+    /// @dev Overridable way to get the weth contract.
+    /// @return token The WETH contract.
+    function _getWethContract()
```

Comment:
I kinda like making this public and making the deployment variables private. That's what we do in the staking contracts.